pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -477,7 +477,20 @@ func applyHTTPRouteDestination(
 	out.Action = &route.Route_Route{Route: action}
 
 	if in.Rewrite != nil {
-		action.PrefixRewrite = in.Rewrite.GetUri()
+		action.ClusterSpecifier = &route.RouteAction_Cluster{
+			Cluster: in.Name,
+		}
+		uri := in.Rewrite.GetUri()
+		if fullURI, isFullPathRewrite := cutPrefix(uri, "*fullreplace*"); isFullPathRewrite && model.UseGatewaySemantics(vs) {
+			action.RegexRewrite = &matcher.RegexMatchAndSubstitute{
```

Comment:
Is this the best only way in envoy to do a full replace ? Can we avoid regex ? If not - please file a ticket on envoy to add full class support, there are other projects implementing K8S Gateway on top of Envoy.

Reply:
There is this `envoy.path.rewrite.uri_template.uri_template_rewriter` [extension](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/path/rewrite/uri_template/v3/uri_template_rewrite.proto#extensions-path-rewrite-uri-template-v3-uritemplaterewriteconfig) but I have not found someone that actually uses it.

seems that envoy gateway are doing the same - https://github.com/envoyproxy/gateway/blob/22131aa1efaa48b414395c5de15097ff99793b7e/internal/xds/translator/route.go#L226-L237